web/client/utils/PluginsUtils.js

```diff
@@ -211,14 +211,25 @@ const isValidConfiguration = (cfg) => {
     return cfg && isString(cfg) || (isObject(cfg) && cfg.name);
 };
 
+const executeDeferredProp = (pluginImpl, pluginConfig, name) => pluginImpl && isFunction(pluginImpl[name]) ?
```

Comment:
what is it needed to? I'd like to have the opinion of @mbarto to check if this can break things or if it is a good improvement

Reply:
This is used in getPluginItems to allow container configuration of plugins to depend on plugin's configuration. This is needed in Export to specify an action in BurgerMenu configuration depending on the value of "enabledFormats" cfg prop of Export plugin. If a container configuration is a function it executes it passing it a plugin configuration before proceeding.

Reply:
It looks good to me. Makes the configuration simpler and could be used in more cases.
There is a system that allow to configure that parts
https://github.com/geosolutions-it/MapStore2/blob/ddd028d29a5dd9e3dad80d94277c0a1bce948566/web/client/utils/PluginsUtils.js#L250

... and a refactor in progress about the plugins system. Therefore I'm asking to @mbarto and/or @allyoucanmap if this can be included or not.

Reply:
@vlt1 @offtherailz looks good to me too, it enables a useful case